src/ixdat/techniques/ec_ms.py

```diff
@@ -426,6 +429,156 @@ def siq_ecms_calibration_curve(
         else:
             return cal
 
+    # DECONVOLUTION METHODS BELOW
+    def grab_deconvoluted_current(
+        self, mol, impulse_response, tspan=None, tspan_bg=None, snr=10
+    ):
+        """Return the deconvoluted partial current for a given signal using the
+        algorithm developed by Krempl et al.
+        https://pubs.acs.org/doi/abs/10.1021/acs.analchem.1c00110
+
+        Note, this actually doesnt need the EC data - it is calculated
+        from calibrated MS data. It is only meaningful for ECMS measurements
+        at the moment, justifying placement here.
+
+        Args:
+            mol (str): Name of molecule for which deconvolution is to
+                be carried out.
+            impulse_response (ECMSImpulseResponse): Kernel object which contains the mass
+                transport parameters
+            tspan (list): Timespan for which the partial current is returned.
+            tspan_bg (list): Timespan that corresponds to the background signal.
+            snr (int): signal-to-noise ratio used for Wiener deconvolution.
+        """
+        # TODO: comments in this method so someone can tell what's going on!
+
+        # grab the calibrated data
+        t_sig, v_sig = self.grab_flux(mol, tspan=tspan, tspan_bg=tspan_bg)
+        # calculate the impulse response
+        signal_response = impulse_response.model_impulse_response_from_params(
+            dt=t_sig[1] - t_sig[0], duration=t_sig[-1] - t_sig[0]
+        )
+        # this seems quite inefficient, to re-calculate the impulse response every time?
+        # TODO: store this somehow
```

Comment:
yes! in the Calculator.

Reply:
There is some preliminary check of this now